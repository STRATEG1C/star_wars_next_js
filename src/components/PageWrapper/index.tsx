import React from 'react';
import Head from 'next/head';

type Props = {
    title: string
}

const PageWrapper: React.FC<Props> = ({ title, children }) => {
    return (
        <div>
            <Head>
                <title>{ title }</title>
                <link rel="icon" href="/favicon.ico" />
            </Head>
            { children }
        </div>
    );
};

export default PageWrapper;
