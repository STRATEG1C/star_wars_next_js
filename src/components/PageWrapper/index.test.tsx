import React from 'react';
import { shallow } from 'enzyme';
import PageWrapper from './index';

describe('Should render PageWrapper and title', () => {
    it('Should render component with title and children', () => {
        const wrapper = shallow(
            <PageWrapper title="Title">
                <p>App</p>
            </PageWrapper>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
