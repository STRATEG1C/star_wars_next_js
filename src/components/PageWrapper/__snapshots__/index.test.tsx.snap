// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render PageWrapper and title Should render component with title and children 1`] = `
<div>
  <Head>
    <title>
      Title
    </title>
    <link
      href="/favicon.ico"
      rel="icon"
    />
    <meta
      content="Test site on Nest js"
      name="description"
    />
    <meta
      content="Next, React, JavaScript"
      name="keywords"
    />
  </Head>
  <p>
    App
  </p>
</div>
`;
